src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/ShapeBreakingDiagnosticSuppressorTests.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using LibraryImportGenerator.UnitTests.Verifiers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.Interop.Analyzers;
+using Xunit;
+
+namespace LibraryImportGenerator.Unit.Tests
+{
+    [ActiveIssue("https://github.com/dotnet/runtime/issues/60650", TestRuntimes.Mono)]
+    public class ShapeBreakingDiagnosticSuppressorTests
+    {
+        [Fact]
+        public async Task StatefulValueMarshallerMethodsThatDoNotUseInstanceState_SuppressesDiagnostic()
+        {
+            await VerifySuppressorAsync("""
+                using System;
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices.Marshalling;
+
+                struct S
+                {
+                    public bool b;
+                };
+
+                [CustomMarshaller(typeof(S), MarshalMode.ManagedToUnmanagedIn, typeof(ManagedToUnmanagedIn))]
+                static class Marshaller
+                {
+                    public struct ManagedToUnmanagedIn
+                    {
+                        public static int BufferSize { get; } = 1;
+
+                        public void {|#0:FromManaged|}(S s)
+                        {
+                        }
```

Comment:
nit: can you one-line these like you did for `Free` and `OnInvoked` to make it a bit more compact?